packages/ERTP/test/unitTests/test-recovery.js

```diff
@@ -0,0 +1,42 @@
+// @ts-check
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { keyEQ, makeCopySet } from '@agoric/store';
+
+import { makeIssuerKit, AmountMath } from '../../src/index.js';
+
+const { isEmpty, isEqual } = AmountMath;
+const emptySet = makeCopySet([]);
+
+test('payment recovery', async t => {
+  const { issuer, mint, brand } = makeIssuerKit('precious');
+  const precious = num => AmountMath.make(brand, num);
+  const payment1 = mint.mintPayment(precious(37n));
+  const payment2 = mint.mintPayment(precious(41n));
+  const alicePurse = issuer.makeEmptyPurse();
+  const bobPurse = issuer.makeEmptyPurse();
+
+  t.assert(keyEQ(alicePurse.getRecoverySet(), emptySet));
+  alicePurse.deposit(payment1);
+  bobPurse.deposit(payment2);
+  t.assert(keyEQ(alicePurse.getRecoverySet(), emptySet));
```

Comment:
Do you also want to assert emptySet on `bobPurse` before and after depositing `payment2`?